packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -1,15 +1,18 @@
 import { z } from "zod";
 
+import { CONFIRM_PASSWORD_KEY } from "../constants/constants.js";
 import { UserValidationMessage, UserValidationRule } from "../enums/enums.js";
 
 type UserSignUpRequestValidationDto = {
+	confirmPassword: z.ZodString;
 	email: z.ZodString;
 	name: z.ZodString;
 	password: z.ZodString;
 };
 
 const userSignUp = z
 	.object<UserSignUpRequestValidationDto>({
+		confirmPassword: z.string().trim(),
```

Comment:
Shouldn't be a part of validation and shouldn't be sent to Backend. You should handle it in sign up onSubmit function